pandas/core/groupby/grouper.py

```diff
@@ -397,10 +395,10 @@ def indices(self):
         return values._reverse_indexer()
 
     @property
-    def labels(self):
-        if self._labels is None:
-            self._make_labels()
-        return self._labels
+    def codes(self):
```

Comment:
If you have a good docstring in mind having looked at this would be very helpful for others

Reply:
ditto on annotations...for any of these functions

Reply:
I typed it up a little, though there's still more to do.